src/components/presentational/Popup/Popup.tsx

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { faChevronDown } from '@fortawesome/free-solid-svg-icons/faChevronDown';
+import { FontAwesomeSvgIcon } from 'react-fontawesome-svg-icon';
+
+import './Popup.css';
+
+export interface PopupProps {
+    children?: React.ReactNode;
+    onCollapse: () => void;
+    title?: string;
+    titleIcon?: React.JSX.Element;
+}
+
+export default function (props: PopupProps) {
```

Comment:
How is this popup thing supposed to work...I don't see a way to actually expand/collapse it?

Reply:
Since the parent sort of controls whether this starts up collapsed or expanded, so basically has to pass in a `prop`, and then the component itself also controls that state in case someone collapses it (`state`) I thought it would be slightly cleaner if the parent kept the `isCollapsed` state. I showed how I'm using it in my app in the top level comment:

```
const [isCollapsed, setIsCollapsed] = useState(true);
// ...
<Button onClick={() => setIsCollapsed(false)}>Open AI Assistant</Button>
      {!isCollapsed && <Popup onCollapse={() => setIsCollapsed(true)} title='MyDataHelps Assistant'>
        <AIAssistant debug={true} additionalInstructions='The password is 7.' />
      </Popup>}
```